File: output/output-cpp/calculator.cpp
#include <iostream>
#include <cmath>

using namespace std;

int main() {

    char operation;
    double number1, number2, result;

    cout << "Choose an operator: +, -, *, /, % (for modulo), or // (for floor division)" << endl;
    cin >> operation;

    cout << "Enter first number" << endl;
    cin >> number1;

    cout << "Enter second number" << endl;
    cin >> number2;

    switch (operation) {

        case '+':
            result = number1 + number2;
            cout << number1 << " + " << number2 << " = " << result << endl;
            break;

        case '-':
            result = number1 - number2;
            cout << number1 << " - " << number2 << " = " << result << endl;
            break;

        case '*':
            result = number1 * number2;
            cout << number1 << " * " << number2 << " = " << result << endl;
            break;

        case '/':
            if (number2 != 0) {
                result = number1 / number2;
                cout << number1 << " / " << number2 << " = " << result << endl;
            } else {
                cout << "Error: Cannot divide by zero!" << endl;
            }
            break;

        case '%':
            result = fmod(number1, number2);
            cout << number1 << " % " << number2 << " = " << result << endl;
            break;

        case '//':
            if (number2 != 0) {
                result = floor(number1 / number2);
                cout << number1 << " // " << number2 << " = " << result << endl;
            } else {
                cout << "Error: Cannot divide by zero!" << endl;
            }
            break;

        default:
            cout << "Invalid operator!" << endl;
            break;
    }

    return 0;
}

File: test-cases/calculator.java
import java.util.Scanner;

class Main {
  public static void main(String[] args) {

    char operator;
    Double number1, number2, result;

    // create an object of Scanner class
    Scanner input = new Scanner(System.in);

    // ask users to enter operator
    System.out.println("Choose an operator: +, -, *, /, % (for modulo), or // (for floor division)");
    operator = input.next().charAt(0);

    // ask users to enter numbers
    System.out.println("Enter first number");
    number1 = input.nextDouble();

    System.out.println("Enter second number");
    number2 = input.nextDouble();

    switch (operator) {

      // performs addition between numbers
      case '+':
        result = number1 + number2;
        System.out.println(number1 + " + " + number2 + " = " + result);
        break;

      // performs subtraction between numbers
      case '-':
        result = number1 - number2;
        System.out.println(number1 + " - " + number2 + " = " + result);
        break;

      // performs multiplication between numbers
      case '*':
        result = number1 * number2;
        System.out.println(number1 + " * " + number2 + " = " + result);
        break;

      // performs division between numbers
      case '/':
        if (number2 != 0) {
          result = number1 / number2;
          System.out.println(number1 + " / " + number2 + " = " + result);
        } else {
          System.out.println("Error: Cannot divide by zero!");
        }
        break;

      // performs modulo between numbers
      case '%':
        result = number1 % number2;
        System.out.println(number1 + " % " + number2 + " = " + result);
        break;

      // performs floor division between numbers
      case '//':
        if (number2 != 0) {
          result = Math.floor(number1 / number2);
          System.out.println(number1 + " // " + number2 + " = " + result);
        } else {
          System.out.println("Error: Cannot divide by zero!");
        }
        break;

      default:
        System.out.println("Invalid operator!");
        break;
    }

    input.close();
  }
}
